pkg/commands/workload_apply.go

```diff
@@ -151,6 +151,12 @@ func (opts *WorkloadApplyOptions) Exec(ctx context.Context, c *cli.Config) error
 		return nil
 	}
 
+	if opts.useLSP(currentWorkload) {
+		if err := checkLSPHealth(ctx, c); err != nil {
```

Comment:
Do we really need to check health explicitly ?  Cant we handle these errors as part of the existing roundtripper itself ?

Reply:
Based on the discussions, since the requirement is to display error message same as that of Health, we would have to call the health endpoint first.